packages/core/src/api/middleware/auth-guard.ts

```diff
@@ -168,4 +180,68 @@ export class AuthGuard implements CanActivate {
         }
         return serializedSession;
     }
+
+    private async getSessionFromToken(
+        req: Request,
+        extracted: ExtractTokenResult,
+        info?: GraphQLResolveInfo,
+    ): Promise<CachedSession | undefined> {
+        if (extracted.method !== 'api-key') {
+            return this.sessionService.getSessionFromToken(extracted.token);
+        }
+
+        const strategy = this.apiKeyService.getApiKeyStrategyByApiType(getApiType(info));
+        const parseResult = strategy.parse(extracted.token);
+        if (!parseResult) {
+            return;
+        }
+
+        const ctx = await this.requestContextService.fromRequest(req, info);
+
+        const apiKey = await this.apiKeyService.findOneByLookupId(ctx, parseResult.lookupId, [
+            'user',
+            'user.roles',
+            'user.roles.channels',
+        ]);
+        if (!apiKey) {
+            return;
+        }
+
+        const isHashMatching = await strategy.hashingStrategy.check(extracted.token, apiKey.apiKeyHash);
+        if (!isHashMatching) {
+            return;
+        }
+
+        this.apiKeyService
```

Comment:
Investiage an alternative solution to make `lastUsedAt` session based, meaning:
- shorter API-Key sessions
- `lastUsedAt` is only updated on session creation (when it expired)

Therefore we have less database write operations in case of API-key usecases that will fire many requests at once or in a short time range.

Reply:
Done! By configuring the newly introduced time interval field, lets say "15 minutes", we only execute the update query once per key per time interval. Theres only one place where we need to update the usage timestamp (auth guard) and there we already have the api key in memory, so we can simply compare the last used date and check for the interval. This may result in drastically fewer db writes.

I added a detailed comment on the api key strategy which explains why and how this works. 

Heres the commit: https://github.com/vendure-ecommerce/vendure/pull/3815/commits/08db7b61455336fd2339a88cd6d76eb04e0816fb